stdlib/src/hashlib/fnv1a.mojo

```diff
@@ -0,0 +1,83 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+
+"""Implements the [Fnv1a 64 bit variant](https://en.wikipedia.org/wiki/Fowler–Noll–Vo_hash_function) algorithm as a Hasher type."""
+
+from .hasher import Hasher, Hashable
+from memory import UnsafePointer
+
+
+struct Fnv1a(Hasher):
```

Comment:
**Suggestion** Worth teasing this and its tests out to a separate PR since this one is blocked on compiler bugs?